### PR TITLE
add version information to JMX metrics & component logs

### PR DESCRIPTION
**Pull request by agavra**
adds logging and JMX metrics of the Pinot Version 

for some reason, I can't get this to work in IntelliJ (but I confirmed that the output works in maven). I looked at some SO [posts](https://stackoverflow.com/questions/34343570/maven-build-is-not-filtering-properties-in-intellij) where others were having similar issues but was not able to resolve it.

image

-----

local testing if you want to repro:

```
mvn resources:resources -pl :pinot-common
```

Then run the classes you want to run and properly resolves the `pinot-version.properties` file

### CODE REVIEW

File: pinot-common/pom.xml [hunk 1]
```diff
@@ -36,6 +36,12 @@
   </properties>
 
   <build>
+    <resources>
+      <resource>
+        <directory>src/main/resources</directory>
+        <filtering>true</filtering>
+      </resource>
+    </resources>
```
Comment (walterddr) [inline pinot-common/pom.xml:44]: what was this for?

File: pinot-common/src/main/resources/pinot-version.properties [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+pinot.version=${project.version}
```
Comment (walterddr) [inline pinot-common/src/main/resources/pinot-version.properties:20]: can we include commit hash (first 6 digit)?
most pinot users follow master closely. it isn't very useful to show `0.12.0-SNAPSHOT`

(but yet it can be address in a separate PR)

**Review by walterddr — APPROVED** [review]
left a couple of questions, otherwise lgtm. tagging @xiangfu0 for another look

File: pinot-common/pom.xml (hunk 1 above)
Comment (agavra) [reply pinot-common/pom.xml:44]: filtering (see the javadoc of `PinotVersion`) is what allows maven to understand that it should replace `${project.version}` with the actual version

File: pinot-common/src/main/resources/pinot-version.properties (hunk 2 above)
Comment (agavra) [reply pinot-common/src/main/resources/pinot-version.properties:20]: I thought about doing that, but it involves either hacking it with bash scripts or pulling in https://github.com/git-commit-id/git-commit-id-maven-plugin (which is not the end of the world, but it's an additional dependency)

Comment (walterddr) [reply pinot-common/src/main/resources/pinot-version.properties:20]: Lets take a look at how some other project handles this. I saw some project produces a version.txt file during the assembly plugin execution and make code read that out from JAR as generated resources

Comment (walterddr) [reply pinot-common/src/main/resources/pinot-version.properties:20]: But yes can be done in separate PR

**Review by siddharthteotia — APPROVED** [review]
LGTM

File: pinot-common/src/main/java/org/apache/pinot/common/version/PinotVersion.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Reads the maven-setting {@code pinot-version.properties} file to extract the
+ * maven version that this code was compiled against.
```
Comment (siddharthteotia) [inline pinot-common/src/main/java/org/apache/pinot/common/version/PinotVersion.java:30]: (nit) a bit misleading I guess. Should be `pinot version`

Comment (agavra) [reply pinot-common/src/main/java/org/apache/pinot/common/version/PinotVersion.java:30]: ah yes.. that javadoc makes little sense reading it back with a fresh mind. I'll fix that up
